src/stream/src/executor/mod.rs

```diff
@@ -373,6 +373,36 @@ impl Barrier {
         }
     }
 
+    /// Whether this barrier adds new downstream actors for the actor with `upstream_actor_id`.
+    pub fn has_new_downstream(&self, upstream_actor_id: ActorId) -> bool {
```

Comment:
This does not seem to be accurate.

For scaling, an actor might have more downstream actors. But it does not count to `dispatcher_num` since one dispatcher is for all actors in the same fragment. So a name like `has_new_downstream_fragment` might be more suitable...?

Then what about the case of `ALTER TABLE sink_into_table`? The table plan is replaced with some **new** fragments to update the schema, but the downstream job number for the upstream sinks will not change. 😄 Although it's not the case of this PR either, for generality maybe `more_downstream_fragment` is better?

Reply:
You are too right